src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/needbraces/InputNeedBracesPatternMatchingForSwitch.java

```diff
@@ -0,0 +1,101 @@
+/*
+NeedBraces
+allowSingleLineStatement = false
+allowEmptyLoopBody = (default)false
+tokens = LITERAL_CASE, LITERAL_DEFAULT, LAMBDA
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.blocks.needbraces;
+
+public class InputNeedBracesPatternMatchingForSwitch {
+
+    void testSwitchStatements(Object o) {
+        switch (o) {
+            case Integer i when (i == 0) :  // violation
+                System.out.println("zero");
+                break;
+
+            case String s when(s.equals("a")) :  // violation
+                System.out.println("a");
+                break;
+
+            case Point(int x, int y ) when(x>=0 && y >=0) :  // violation
+                System.out.println("point");
+                break;
+            default :  // violation
+                System.out.println("default");
+        }
+
+         switch (o) {
+            case Integer i when (i == 0) : {
+                System.out.println("zero");
+                break;
+            }
+            case String s when(s.equals("a")) : {
+                System.out.println("a");
+                break;
+            }
+            case Point(int x, int y ) when(x>=0 && y >=0) : {
```

Comment:
```suggestion
            case Point(int x, int y ) when (x>=0 && y >=0) : {
```
Please fix all of these

Reply:
done